Libs/MRML/Core/vtkMRMLMessageCollection.h

```diff
@@ -0,0 +1,98 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#ifndef __vtkMRMLMessageCollection_h
+#define __vtkMRMLMessageCollection_h
+
+// MRML includes
+#include "vtkMRML.h"
+
+// VTK includes
+#include <vtkIdTypeArray.h>
+
+// STD includes
+#include <string>
+#include <vector>
+
+///
+/// A class for recording warnings and errors associated with this
+/// vtkMRMLStorableNode.  A substantially similar vtkMessageCollection
+/// is planned for vtkAddon.  When that is released, this will be
+/// replaced by that.
+class VTK_MRML_EXPORT vtkMRMLMessageCollection : public vtkObject
+{
+public:
+  vtkTypeMacro(vtkMRMLMessageCollection,vtkObject);
+  void PrintSelf(ostream& os, vtkIndent indent) override;
+  static vtkMRMLMessageCollection* New();
+
+  ///
+  /// A helper class that describes a single message
+  struct Message
+  {
+    Message(unsigned long messageType, const std::string &messageText);
+    ///
```

Comment:
Adding a function object perhaps more elegant, but it means 20+ lines of extra code to save 2-3 lines of code at a one place (in GetNumberOfMessagesOfType). If you want to be fancy then use a lambda function, then at least the code would not be much longer (or even 1-2 lines shorter) than doing a trivial iteration through the list).

Reply:
I eliminated the `EqualMessageType` function object and went with the "2-3 lines of code" in `GetNumberOfMessagesOfType`.